strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ArtifactCoordinatesService.java

```diff
@@ -47,6 +50,20 @@ protected boolean identifyEntity(AbstractArtifactCoordinates entity)
         return true;
     }
 
+    @Override
+    protected <S extends AbstractArtifactCoordinates> S cascadeEntitySave(AbstractArtifactCoordinates entity)
+    {
+        try
+        {
+            return super.cascadeEntitySave(entity);
+        }
+        catch (OConcurrentModificationException ex)
```

Comment:
why this needed?
I thought we already fixed all of such issues with `AritfactEntry` and `ArtifactCoordinates`.

Reply:
see [here](https://jenkins.carlspring.org/job/strongbox/job/strongbox/job/PR-987/13/testReport/)

Short explanation: when we save artifact group [here](https://github.com/strongbox/strongbox/pull/987/files#diff-ca2af8f423f4756b1737ad2178c1101dR85) all related `artifactEntries` and `artifactCoordinates` have updated `entityVersion`.

Reply:
This is related with `org.carlspring.strongbox.data.tx.OEntityUnproxyAspect` and `com.orientechnologies.orient.core.annotation.OVersion`

Reply:
this should not happen because we have synchronization by `ArtifactCoordinates.getId()` [here](https://github.com/strongbox/strongbox/blob/SB-1253/strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/RepositoryPathLock.java#L66).

can you please double check it?

Reply:
[Here](https://jenkins.carlspring.org/job/strongbox/job/strongbox/view/change-requests/job/PR-987/16/) is the PR build with this catch.

Reply:
And [here](https://jenkins.carlspring.org/job/strongbox/job/strongbox/job/PR-987/17/) is without this catch. I'll try to analyze it deeper.

Reply:
The reason is that when we fetch the `ArtifactGroup`, all related `artifactEntries` are eagerly fetched too (`@OneToMany(fetch = FetchType.LAZY)` does not work in our case in orientDb :( ). Inside these fetched collection of `artifactEntries`  there might be cases with the same `artifactCoordinates` (let say artifactEntry1 and artifactEntry2). If we want to remove the `last-version` tag from  `artifactEntry1` first and then from `artifactEntry2` we've got `OConcurrentModificationException ` because they share the same `artifactCoordinates` .

Reply:
I also tried https://orientdb.com/docs/last/Fetching-Strategies.html without luck. It doesn't work in our case.

Reply:
It can also be solved like [this](https://github.com/strongbox/strongbox/pull/987/commits/8ec7ebbfc69f722fd5df46b7d3f78bf952b18d7d).

Reply:
And build is green with it. What do you think, @sbespalov ?

Reply:
@fuss86 I suggest to leave concurrency and make `last-version` update work as expected first.
I think we have some confusions about it.

Reply:
No longer valid.

Reply:
:+1: